### PR TITLE
binloginfo: add default charset and collate in binlog output [ DNM ]

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve? 
fix  [TIDB-4108](https://internal.pingcap.net/jira/browse/TIDB-4108) binlog output without default charset and collate

### What is changed and how it works?

- There is  a func `SetDDLBinlog` in binloginfo.go which puts the orgin ddl string from user into binlog info. 
- So I append the ddl string with the default charset and collate before it's put into binlog info.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - add two sub func `reverseString` and `appendDefaultCharsetAndCollate` to reverse string and append string.
- add append logic in `SetDDLBinlog`.

### CODE REVIEW

File: sessionctx/binloginfo/binloginfo.go [hunk 1]
```diff
@@ -14,6 +14,7 @@
 package binloginfo
 
 import (
+	"github.com/pingcap/parser/model"
```
Comment (crazycs520) [inline sessionctx/binloginfo/binloginfo.go]: Please put this line in suitable location.

File: ddl/ddl_worker.go [hunk 2]
```diff
@@ -419,7 +419,7 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 				return errors.Trace(err)
 			}
 			if job.IsDone() || job.IsRollbackDone() {
-				binloginfo.SetDDLBinlog(d.binlogCli, txn, job.ID, job.Query)
+				binloginfo.SetDDLBinlog(d.binlogCli, txn, job, job.Query)
```
Comment (crazycs520) [inline ddl/ddl_worker.go]: Look like the parameter of `job.Query` is redundant now.

File: sessionctx/binloginfo/binloginfo.go [hunk 3]
```diff
@@ -131,11 +132,27 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
 }
 
 // SetDDLBinlog sets DDL binlog in the kv.Transaction.
-func SetDDLBinlog(client *pumpcli.PumpsClient, txn kv.Transaction, jobID int64, ddlQuery string) {
+func SetDDLBinlog(client *pumpcli.PumpsClient, txn kv.Transaction, job *model.Job, ddlQuery string) {
 	if client == nil {
 		return
 	}
 
+	//append default charset and collate
```
Comment (crazycs520) [inline sessionctx/binloginfo/binloginfo.go]: How about extract below logic to a single function and add a unit test for the function.

File: sessionctx/binloginfo/binloginfo_test.go [hunk 4]
```diff
@@ -147,6 +147,30 @@ func (s *testBinlogSuite) TestBinlog(c *C) {
 	}
 	c.Assert(matched, IsTrue)
 
+	tk.MustExec("drop table if exists t1")
```
Comment (crazycs520) [inline sessionctx/binloginfo/binloginfo_test.go]: please add test for the partition table, eg:
```sql
create table t2 (a int,b int,primary key(a,b)) partition by hash(a) partitions 4;
create table t3 (a int key) partition by range(a) (
		partition p0 values less than (10),
		partition p1 values less than (20)
	);
```

File: sessionctx/binloginfo/binloginfo.go (hunk 1 above)
Comment (AilinKid) [reply sessionctx/binloginfo/binloginfo.go]: Roger that!

File: sessionctx/binloginfo/binloginfo.go (hunk 3 above)
Comment (AilinKid) [reply sessionctx/binloginfo/binloginfo.go]: Roger that!

File: sessionctx/binloginfo/binloginfo_test.go (hunk 4 above)
Comment (AilinKid) [reply sessionctx/binloginfo/binloginfo_test.go]: I don't take care of this case, thx!

**Pull request closed without merging**